### ISSUE TITLE
Use toolchain information if available

### ISSUE THREAD

**Issue by ljnelson** [payara/ecosystem-maven#18]
Hi; thanks for your work on Payara!

I've found that I wish this plugin were [toolchain aware](https://maven.apache.org/guides/mini/guide-using-toolchains.html).

I have Java 9 set up as my default `java`, but my `toolchains.xml` file specifies Java 8.  Tools like the compiler, javadoc, etc. automatically use this information since I make use of the `maven-toolchain-plugin`.  It would be nice if this plugin did too.

**Comment by mulderbaba** [comment]
@ljnelson sure, a profile like given below would help you on this? I can define it w/ a profile named ``toolchain``.
```
        <profile>
            <id>toolchain</id>
            <build>
                <plugins>
                    <plugin>
                        <groupId>org.apache.maven.plugins</groupId>
                        <artifactId>maven-toolchains-plugin</artifactId>
                        <version>1.1</version>
                        <executions>
                            <execution>
                                <phase>validate</phase>
                                <goals>
                                    <goal>toolchain</goal>
                                </goals>
                            </execution>
                        </executions>
                        <configuration>
                            <toolchains>
                                <jdk>
                                    <version>[1.8]</version>
                                </jdk>
                            </toolchains>
                        </configuration>
                    </plugin>
                </plugins>
            </build>
        </profile>
```

**Comment by ljnelson** [comment]
With respect, no, not at all.

For there to be toolchain support in a plugin, the plugin author must change her code to make use of internal Maven toolchain classes.  For example, here's a link to the `maven-compiler-plugin` and how it makes use of the `jdk` toolchain settings: https://github.com/apache/maven-compiler-plugin/blob/master/src/main/java/org/apache/maven/plugin/compiler/CompilerMojo.java#L213-L217  Or here's how the `exec-maven-plugin` does it: https://github.com/mojohaus/exec-maven-plugin/blob/9c99ac931af4f406f8733cdebbc7932c0d28c54c/src/main/java/org/codehaus/mojo/exec/ExecMojo.java#L639-L661

Thanks for taking the time to look at this issue!

**Comment by mulderbaba** [comment]
Internal issue: MP-18

**Comment by mulderbaba** [comment]
Closed by #60
